include/oneapi/dpl/pstl/algorithm_fwd.h

```diff
@@ -247,6 +247,19 @@ __pattern_walk3(_ExecutionPolicy&& __exec, _RandomAccessIterator1 __first1, _Ran
                 _RandomAccessIterator2 __first2, _RandomAccessIterator3 __first3, _Function __f, _IsVector __is_vector,
                 /*parallel=*/::std::true_type);
 
+template <class _ForwardIterator1, class _ForwardIterator2, class _ForwardIterator3, class _Function, class _Predicate>
+_ForwardIterator3
+__brick_mask_walk3(_ForwardIterator1 __first1, _ForwardIterator1 __last1, _ForwardIterator2 __first2,
```

Comment:
A minor comment regarding patterns/brick names - I think suffix should be 'mask_walk2', because it is a walker which calls a callable object with 2 arguments(not 3), by mask.